### PR TITLE
Add log when calendar is deleted

### PR DESCRIPTION
**Pull request by kosssi**

### CODE REVIEW

**Comment by kosssi** [comment]
@nono if you can review this PR ;) thanks

File: src/app/replicator/main.coffee [hunk 1]
```diff
@@ -471,29 +471,35 @@ module.exports = class Replicator extends Backbone.Model
...
-            # todo: find better solution
             if err?.status is 404
-                console.info 'The above 404 is not normal, but we retest the \
-                              synchronisation after.'
-                log.warn err
-                return callback()
+                console.info 'The above 404 is not normal, but we retest with \
+                              smallest checkpoint.'
+
+                @checkpointLoop = 0 unless @checkpointLoop
```
Comment (nono) [inline src/app/replicator/main.coffee]: Why not

```coffee
@checkpointLoop ?= 0
```

?

File: src/app/replicator/main.coffee [hunk 2]
```diff
@@ -471,29 +471,35 @@ module.exports = class Replicator extends Backbone.Model
...
+                @checkpointLoop++
+
+                unless @checkpointLoop > 10
+                    options.remoteCheckpoint--
+                    return @sync options, callback
+                else
+                    callback()
```
Comment (nono) [inline src/app/replicator/main.coffee]: I don't like `unless ... else ...`

File: src/app/replicator/main.coffee [hunk 3]
```diff
@@ -471,29 +471,35 @@ module.exports = class Replicator extends Backbone.Model
...
-                log.warn err
-                return callback()
+                console.info 'The above 404 is not normal, but we retest with \
+                              smallest checkpoint.'
+
+                @checkpointLoop = 0 unless @checkpointLoop
+                @checkpointLoop++
+
+                unless @checkpointLoop > 10
+                    options.remoteCheckpoint--
```
Comment (nono) [inline src/app/replicator/main.coffee]: Why `--` and not `++`?

What if `options.remoteCheckpoint` became negative?

File: src/app/replicator/replication_launcher.coffee [hunk 4]
```diff
@@ -41,7 +41,7 @@ module.exports = class ReplicationLauncher
     start: (options, callback = ->) ->
         log.debug "start"
 
-        unless @replication
+        if !@replication and @config.get('appState') is 'launch'
```
Comment (nono) [inline src/app/replicator/replication_launcher.coffee]: Cozy guidelines prefer `not` over `!`

File: src/app/views/config.coffee [hunk 5]
```diff
@@ -48,7 +48,7 @@ module.exports = class ConfigView extends BaseView
     formatDate: (date) ->
         log.debug "formatDate #{date}"
 
-        return t 'never' unless date or date is ''
+        return t 'never' unless date or date isnt ''
```
Comment (nono) [inline src/app/views/config.coffee]: I think it should be an `and`, not an  `or` in the conditional. But you should write it with an `if`, not an `unless` to make it easier to reason about.

File: src/app/views/login.coffee [hunk 6]
```diff
@@ -48,8 +48,9 @@ module.exports = class LoginView extends BaseView
 
     getRenderData: ->
         @config = window.app.init.config
+        password = if @inputPassword then @inputPassword.val() else ''
```
Comment (nono) [inline src/app/views/login.coffee]: Maybe

```coffee
password = @inputPassword?.val() or ''
```

File: src/app/replicator/main.coffee (hunk 3 above)
Comment (kosssi) [inline src/app/replicator/main.coffee]: options.remoteCheckpoint is couchdb last_seq ;)

**Comment by kosssi** [comment]
All is done.

File: src/app/replicator/main.coffee [hunk 7]
```diff
@@ -471,29 +471,35 @@ module.exports = class Replicator extends Backbone.Model
...
+
+                @checkpointLoop ?= 0
+                @checkpointLoop++
+
+                if @checkpointLoop > 10
+                    return callback()
+                else
+                    options.remoteCheckpoint--
+                    return @sync options, callback
+
```
Comment (nono) [inline src/app/replicator/main.coffee]: You have a return in both branches (then and else). So the `callback err` below is never called

File: src/app/replicator/main.coffee (hunk 3 above)
Comment (nono) [inline src/app/replicator/main.coffee]: Yes, it is. But it's not my question.

File: src/app/replicator/main.coffee (hunk 7 above)
Comment (kosssi) [inline src/app/replicator/main.coffee]: no it's call when error status isn't 404 ;)

Comment (nono) [inline src/app/replicator/main.coffee]: Okay, my bad.

**Pull request merged**